Title: IBM SFTP server generates permission errors in the Mailbox.log and on the Communication Sessions screen when SFTP client put - United States

Text:
STERLINGNFX TECHNOTE (TROUBLESHOOTING)

PROBLEM(ABSTRACT)
 SFTP server generates permission errors in the Mailbox.log and on the Communication Sessions screen when SFTP client put 

SYMPTOM


This problem only occurs with certain SFTP Clients. It appears to occur with a client that appends the userid/Mailbox prior to the file name that you intent to store onto GIS SFTP server. It's been recreated using CoreFTP Client SSH/SFTP. I was unable to recreate the problem with WS FTP Pro client because it does not append the userid/Mailbox prior to the file name when it stores/put a file onto GIS SFTP server. During recreation for example, I selected to stor a filename of YServer.txt, but the CoreFTP SFTP Client sends GIS a filename of /Tanya/YServer.txt, I would have expected the client to only send me /YServer.txt. Customer running at GIS 4.1.1 Build Version: 1980_BR4770095_SR1366739_1368361.

 

In house test results at GIS 4.1.1 Build 1990, GIS 4.2 Build 4217 and GIS 4.3 Build 4317: GIS does successfully put/store the file, but it also post permission denial error messages on the Communication Session Detail screen and in the Mailbox.log. Also the Communication Session Detail screen has a Status of error in red. In my test, focusing on stored filename YServer.txt, it was successfully put/stored (146 bytes), but only after being previously denied twice with error message “Mailbox path does not exist or authorization is denied.” being posted. The denial and access of the store all occurs within the same second. The Mailbox.log post a 'User does not have mailbox permission error'.

Error Message  

 

 

The corresponding Mailbox.log post a 'User does not have mailbox permission error' at bottom:

[2009-02-20 17:39:27.351] DEBUG 000000000000 GLOBAL_SCOPE PermissionManager.haveMailboxPermission() entered mailboxPath:/Tanya user:Tanya 

[2009-02-20 17:39:27.351] DEBUG 000000000000 GLOBAL_SCOPE PermissionManager.haveMailboxPermission() entered mailboxPath:/Tanya user:Tanya: bypass virtual root comparision 

[2009-02-20 17:39:27.351] DEBUG 000000000000 GLOBAL_SCOPE PermissionManager.haveMailboxPermission() <<<Permission Exists>>> for /Tanya user:Tanya 

[2009-02-20 17:39:27.351] DEBUG 000000000000 GLOBAL_SCOPE PermissionManager.haveMailboxPermission() exiting mailboxPath:/Tanya user:Tanya 

[2009-02-20 17:39:27.352] DEBUG 000000000000 GLOBAL_SCOPE PermissionManager.haveMailboxPermission() entered mailboxPath:/Tanya/test.txt user:Tanya 

[2009-02-20 17:39:27.355] ALL 000000000000 GLOBAL_SCOPE PermissionManager.haveMailboxPermission() User does not have mailbox permission:/Tanya/test.txt.mbx mailboxPath:/Tanya/test.txt user:Tanya 

 

The Communication Sessions Detail screen post error message “Mailbox path does not exist or authorization is denied.” And the STATUS is post 'ERROR' in red. 


RESOLVING THE PROBLEM


 

 

 

 Development finds the GIS FTP Server is working as designed and believes this is a CoreFTP Client issue. 

 We find the behavior seems to be quite strange for CoreFTP when compared 

 with other SFTP Clients. 

 The Error message "Mailbox path does not exist or authorization denied" is" 

 is supposedly displayed when the client try to get the file from the server 

 which doesn't exists in the mailbox. It is valid in this case. 

 However, when trying to put a file into SFTP Server, CoreFTP Client first 

 checks whether the file exists in the mailbox or not. Since the file 

 doesn't already exist in the mailbox (first time being put) GIS first 

 displays the Error message and also records the event in the Communication 

 Sessions screen and then put the file onto mailbox. We see no issue with 

 GIS SFTP Server, this check is being done only by CoreFTP; hence it is a 

 client's issue. 

 

 

HISTORICAL NUMBER
 NFX5174 

PRODUCT ALIAS/SYNONYM
 

Function Area


Mailbox


Escalation ID


4981924


Severity


Normal


Type


NormalFix